src/cmake.ts

```diff
@@ -464,17 +456,24 @@ class ErrorParser extends util.OutputParser {
     private _activeParser: Maybe<DiagnosticParser>;
     private _parserCollection: Set<DiagnosticParser>;
 
-    constructor(binaryDir: string) {
+    constructor(binaryDir: string, parsers: Maybe<string[]>) {

```

Comment:
Also, maybe a literal type to represent the names of parsers, rather than any arbitrary string. Like:

~~~typescript
type ParserID = ('cmake', 'gcc', 'gnuld', 'msvc', 'ghs');
~~~

Then the compiler would even be able to check for typos.